tests/app/main/test_suppliers.py

```diff
@@ -1665,3 +1666,192 @@ def test_should_show_email_address_even_when_refreshed(self):
+    ))
+    @mock.patch("app.main.views.suppliers.DMMailChimpClient")
+    def test_post_invalid_email(self, mailchimp_client_class, email_address_value, expected_validation_message):
+        mailchimp_client_instance = mock.Mock(spec=("subscribe_new_email_to_list",))
+        mailchimp_client_instance.subscribe_new_email_to_list.side_effect = AssertionError("This should not be called")
+
+        mailchimp_client_class.side_effect = assert_args_and_return(
+            mailchimp_client_instance,
+            "not_a_real_username",
+            "not_a_real_key",
+            mock.ANY,
+        )
+
+        with self.client as c:
+            response = c.post(
+                "/suppliers/mailing-list",
+                data={
+                    "email_address": email_address_value,
+                },
+            )
+            assert response.status_code == 400
+            doc = html.fromstring(response.get_data(as_text=True), base_url="/suppliers/mailing-list")
+
+            form = self._common_page_asserts_and_get_form(doc)
+            assert tuple(inp.value for inp in form.xpath(".//input[@name='email_address']")) == (
+                email_address_value,
+            )
+
+            assert doc.xpath(
+                "//label[@for='input-email_address']//*[contains(@class, 'validation-message')]"
+                "[normalize-space(string())=$t]",
+                t=expected_validation_message,
+            )
+
+            with self.client.session_transaction() as session:
+                assert not session.get('_flashes')
+
+    @pytest.mark.parametrize("mc_retval", (True, False,))
+    @mock.patch("app.main.views.suppliers.DMMailChimpClient")
+    def test_post_valid_email_failure(self, mailchimp_client_class, mc_retval):
+        mailchimp_client_instance = mock.Mock(spec=("subscribe_new_email_to_list",))
+        mailchimp_client_instance.subscribe_new_email_to_list.side_effect = assert_args_and_return(
+            mc_retval,
+            "not_a_real_mailing_list",
+            "squinting@ger.ty",
+        )
+
+        mailchimp_client_class.side_effect = assert_args_and_return(
+            mailchimp_client_instance,
+            "not_a_real_username",
+            "not_a_real_key",
+            mock.ANY,
+        )
+
+        with self.client as c:
+            response = c.post(
+                "/suppliers/mailing-list",
+                data={
+                    "email_address": "squinting@ger.ty",
+                },
+            )
+            assert response.status_code == 503
+            doc = html.fromstring(response.get_data(as_text=True), base_url="/suppliers/mailing-list")
+
+            assert mailchimp_client_instance.subscribe_new_email_to_list.called is True
+
+            form = self._common_page_asserts_and_get_form(doc)
+            assert tuple(inp.value for inp in form.xpath(".//input[@name='email_address']")) == (
+                "squinting@ger.ty",
```

Comment:
I know `.ty` isn't a valid top level domain, but we generally try to use `@example.com` for fake emails in tests, as in "in case"

Reply:
That does kinda limit you when you're trying to test an email address validator.

Reply:
Ha!  True.